### PR TITLE
Updated the Dockerfile to use python 3.8

### PR DESCRIPTION
**Pull request by mraarif**
this PR is a reattempt of #196 that got reverted.

### CODE REVIEW

**Comment by mraarif** [comment]
I was able to build an image after adding `build-essential` (as suggested by @jmbowman ) to fix the issue we had in the last PR(#196 )

**Comment by natabene** [comment]
@mraarif Is this ready? Does this need a review?

**Comment by mraarif** [comment]
yes, it should be ready for a re-review, @natabene

File: Dockerfile [hunk 1]
```diff
@@ -25,10 +25,14 @@ RUN apt-get update && \
     apt-get install -y software-properties-common && \
     apt-add-repository -y ppa:deadsnakes/ppa && \
     apt-get update && apt-get upgrade -qy && \
-    apt-get install language-pack-en locales git python3.6 python3-pip libmysqlclient-dev libssl-dev python3-dev python3.8-dev python3.8-distutils -qy && \
-    pip3 install --upgrade pip setuptools && \
+    apt-get install language-pack-en locales git libmysqlclient-dev libssl-dev build-essential python3.8-dev python3.8-venv -qy && \
```
Comment (adzuci) [inline Dockerfile:28]: @mraarif Would you be able to split this apt-get line into multiple lines like this:  https://github.com/edx/edx-cookiecutters/blob/master/cookiecutter-django-ida/%7B%7Bcookiecutter.repo_name%7D%7D/Dockerfile#L20-L29

https://openedx.atlassian.net/wiki/spaces/SRE/pages/1838350349/Dockerfile+Best+Practices

File: Dockerfile [hunk 2]
```diff
@@ -44,9 +48,11 @@ WORKDIR /edx/app/notes
 # Copy the requirements explicitly even though we copy everything below
 # this prevents the image cache from busting unless the dependencies have changed.
 COPY requirements/base.txt /edx/app/notes/requirements/base.txt
+COPY requirements/pip.txt /edx/app/notes/requirements/pip.txt
 
 # Dependencies are installed as root so they cannot be modified by the application user.
-RUN pip3 install -r requirements/base.txt
+RUN pip install -r requirements/pip.txt
```
Comment (adzuci) [inline Dockerfile:54]: @mraarif Could you explain how pip can install pip and why we need this extra file or why we don't want to install pip the way we do in the cookiecutter? https://github.com/edx/edx-cookiecutters/blob/master/cookiecutter-django-ida/%7B%7Bcookiecutter.repo_name%7D%7D/Dockerfile#L24

File: Dockerfile (hunk 1 above)
Comment (adzuci) [reply Dockerfile:28]: You may also want to update the comment lines above this this RUN command to match your changes.

File: requirements/pip.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+pip==20.2.3
```
Comment (adzuci) [inline requirements/pip.txt:1]: Do we have a pip.txt file in other services?

**Review by adzuci — CHANGES_REQUESTED** [review]
I put a few questions in your PR.

File: Dockerfile (hunk 1 above)
Comment (mraarif) [reply Dockerfile:28]: sure

File: Dockerfile (hunk 2 above)
Comment (mraarif) [reply Dockerfile:54]: I believe cookiecutter is still using Dockerfile that installs python 3.5 and when we install `python3.8-dev` and `python3.8-venv` `pip` gets installed with them, this `pip install pip` is here to ensure a specific version of pip gets installed across services
ref: https://github.com/edx/registrar/pull/354#discussion_r497710581

File: requirements/pip.txt (hunk 3 above)
Comment (mraarif) [reply requirements/pip.txt:1]: we've started to introduce this across services with python 3.8 update for Dockerfile in services
ref: https://github.com/edx/registrar/pull/354#discussion_r497710581

**Comment by adzuci** [comment]
I also added @tuchfarber as an optional reviewer since he may be able to offer some insight on how we are installing pip in other Dockerfiles.

**Review by adzuci — APPROVED** [review]
Looks good to me.

**Comment by edx-deployment** [comment]
GitHub Actions has started building this code into a docker file.  Check status at https://github.com/edx/edx-notes-api/actions/runs/291768687

**Comment by edx-deployment** [comment]
A docker container including this PR has been built and shipped to docker hub.  Check it out at https://hub.docker.com/r/openedx/edx-notes-api/tags?page=1&name=26b9e33f57432340d049f1635d310398ce8272dd

**Pull request merged**